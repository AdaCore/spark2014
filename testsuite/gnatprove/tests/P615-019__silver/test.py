from test_support import *

# Do not use alt-ergo which seems to be unsound on this test

prove_all(prover=["cvc4", "z3"])
