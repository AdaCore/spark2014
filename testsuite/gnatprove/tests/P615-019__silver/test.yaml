prove_all:
  counterexample: False
  enable_sarif_check: True
large: True
