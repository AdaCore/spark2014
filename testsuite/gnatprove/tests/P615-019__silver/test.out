contracts.adb:3:04: warning: analyzing unreferenced procedure "Convert_Integer0"
contracts.adb:3:45: info: initialization of "U" proved
contracts.adb:5:12: medium: range check might fail (e.g. when X = -1)
contracts.adb:8:44: info: initialization of "U" proved
contracts.adb:12:12: info: range check proved (CVC4: 1 VC)
contracts.adb:15:45: info: initialization of "U" proved
contracts.adb:17:19: info: postcondition proved (CVC4: 1 VC)
contracts.adb:17:23: info: initialization of "U" proved
contracts.adb:20:12: info: range check proved (CVC4: 1 VC)
contracts.adb:23:04: warning: analyzing unreferenced procedure "Call_Convert_Integer"
contracts.adb:26:07: info: precondition proved (CVC4: 1 VC)
contracts.adb:27:14: info: overflow check proved (CVC4: 1 VC)
contracts.adb:27:14: medium: range check might fail
contracts.adb:27:16: info: initialization of "Z" proved
contracts.adb:30:04: warning: analyzing unreferenced procedure "Call_Convert_Integer1"
contracts.adb:33:07: info: precondition proved (CVC4: 1 VC)
contracts.adb:34:14: info: overflow check proved (CVC4: 1 VC)
contracts.adb:34:14: info: range check proved (CVC4: 1 VC)
contracts.adb:34:16: info: initialization of "Z" proved
discriminant_check.adb:12:04: warning: analyzing unreferenced function "Get_X"
discriminant_check.adb:12:13: info: range check proved (CVC4: 1 VC)
discriminant_check.adb:14:15: medium: discriminant check might fail (e.g. when R = (X => 0, others => ?))
discriminant_check.adb:17:04: warning: analyzing unreferenced procedure "Set_X"
discriminant_check.adb:19:08: medium: discriminant check might fail
division_by_zero.adb:9:04: warning: analyzing unreferenced procedure "Oper_Integer"
division_by_zero.adb:9:55: info: initialization of "U" proved
division_by_zero.adb:12:27: medium: divide by zero might fail (e.g. when Y = 0)
division_by_zero.adb:12:27: medium: overflow check might fail (e.g. when X = -1 and Y = 0)
division_by_zero.adb:13:27: medium: divide by zero might fail (e.g. when Y = 0)
division_by_zero.adb:14:27: medium: divide by zero might fail (e.g. when Y = 0)
division_by_zero.adb:18:04: warning: analyzing unreferenced procedure "Oper_Unsigned"
division_by_zero.adb:18:57: info: initialization of "U" proved
division_by_zero.adb:21:27: medium: divide by zero might fail (e.g. when Y = 0)
division_by_zero.adb:22:27: medium: divide by zero might fail (e.g. when Y = 0)
division_by_zero.adb:23:27: medium: divide by zero might fail (e.g. when Y = 0)
division_by_zero.adb:27:04: warning: analyzing unreferenced procedure "Div_Float"
division_by_zero.adb:27:39: info: initialization of "U" proved
division_by_zero.adb:29:14: medium: divide by zero might fail
division_by_zero.adb:29:14: medium: float overflow check might fail
division_by_zero.adb:32:04: warning: analyzing unreferenced procedure "Div_Fixed"
division_by_zero.adb:32:39: info: initialization of "U" proved
division_by_zero.adb:34:14: medium: divide by zero might fail
division_by_zero.adb:34:14: medium: overflow check might fail
division_by_zero.adb:37:04: warning: analyzing unreferenced procedure "Exp_Float"
division_by_zero.adb:37:49: info: initialization of "U" proved
division_by_zero.adb:39:14: medium: divide by zero might fail (e.g. when Y = 0)
division_by_zero.adb:39:14: medium: float overflow check might fail
index_check.adb:3:04: warning: analyzing unreferenced function "Get"
index_check.adb:5:16: medium: array index check might fail (e.g. when J = 1 and S = (others => 'NUL') and S'First = 2 and S'Last = 0)
index_check.adb:8:04: warning: analyzing unreferenced procedure "Set"
index_check.adb:10:09: medium: array index check might fail (e.g. when J = 1 and S'First = 2)
overflow_check.adb:9:04: warning: analyzing unreferenced procedure "Oper_Integer"
overflow_check.adb:9:68: info: initialization of "U" proved
overflow_check.adb:12:30: medium: overflow check might fail (e.g. when X = -2147483648)
overflow_check.adb:13:30: medium: overflow check might fail (e.g. when X = -2147483648)
overflow_check.adb:14:32: medium: overflow check might fail (e.g. when X = -2147483648 and Y = -1)
overflow_check.adb:15:32: medium: overflow check might fail (e.g. when X = -2147483648 and Y = 1)
overflow_check.adb:16:32: medium: overflow check might fail (e.g. when X = -2 and Y = 2)
overflow_check.adb:17:32: medium: divide by zero might fail (e.g. when Y = 0)
overflow_check.adb:17:32: medium: overflow check might fail (e.g. when X = -1 and Y = 0)
overflow_check.adb:18:32: medium: overflow check might fail (e.g. when E = 0 and X = 0)
overflow_check.adb:22:04: warning: analyzing unreferenced procedure "Oper_Float"
overflow_check.adb:22:64: info: initialization of "U" proved
overflow_check.adb:27:32: medium: float overflow check might fail
overflow_check.adb:28:32: medium: float overflow check might fail
overflow_check.adb:29:32: medium: float overflow check might fail
overflow_check.adb:30:32: medium: divide by zero might fail
overflow_check.adb:30:32: medium: float overflow check might fail
overflow_check.adb:31:32: info: division check proved (CVC4: 2 VC)
overflow_check.adb:31:32: medium: float overflow check might fail
overflow_check.adb:35:04: warning: analyzing unreferenced procedure "Oper_Fixed"
overflow_check.adb:35:51: warning: unused variable "E"
overflow_check.adb:35:64: medium: "U" might not be initialized in "Oper_Fixed"
overflow_check.adb:38:30: medium: overflow check might fail
overflow_check.adb:39:30: medium: overflow check might fail
overflow_check.adb:40:32: medium: overflow check might fail
overflow_check.adb:41:32: medium: overflow check might fail
overflow_check.adb:42:32: medium: overflow check might fail
overflow_check.adb:43:32: medium: divide by zero might fail
overflow_check.adb:43:32: medium: overflow check might fail
range_check.adb:16:04: warning: analyzing unreferenced procedure "Convert_Enum"
range_check.adb:16:38: info: initialization of "U" proved
range_check.adb:18:12: medium: range check might fail (e.g. when X = A)
range_check.adb:21:04: warning: analyzing unreferenced procedure "Convert_Integer"
range_check.adb:21:44: info: initialization of "U" proved
range_check.adb:23:12: medium: range check might fail (e.g. when X = -1)
range_check.adb:26:04: warning: analyzing unreferenced procedure "Convert_Unsigned"
range_check.adb:26:46: info: initialization of "U" proved
range_check.adb:28:12: medium: range check might fail (e.g. when X = 11)
range_check.adb:31:04: warning: analyzing unreferenced procedure "Convert_Float"
range_check.adb:31:40: info: initialization of "U" proved
range_check.adb:33:12: medium: range check might fail
range_check.adb:36:04: warning: analyzing unreferenced procedure "Convert_Fixed"
range_check.adb:36:40: info: initialization of "U" proved
range_check.adb:38:12: medium: range check might fail
